### PR TITLE
[C3] Skip empty summary rows

### PR DESCRIPTION
**Pull request by DaniFoldi**
## What this PR solves / how to test

Hi :wave:,

Before getting side-tracked with #5482, I found that when using `.` as the project location, there was nowhere to `cd` to, so the first line in the next steps output summary was skipped. Instead of being properly skipped, `undefined undefined` was printed (see screenshot). This change should skip over these lines where a step is excluded.

image

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [ ] Included
  - [x] Not necessary because: Didn't see any tests for the summary
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): <https://github.com/cloudflare/cloudflare-docs/pull/>...
  - [x] Not necessary because: bugfix

### CODE REVIEW

**Comment by dario-piotrowicz** [comment]
Hi @DaniFoldi 😄👋 

Thanks a lot for the fix 🙂 

but I feel like the change is not the right solution, I think that instead of filtering the thing out we should just not add it in the summary in the first place

I think that this code is slightly off:
https://github.com/cloudflare/workers-sdk/blob/940ad89713fa086f23d394570c328716bfb1bd59/packages/create-cloudflare/src/summary.ts#L15-L17

And changing it in the following way should fix the issue:
```diff
 
        const dirRelativePath = relative(ctx.originalCWD, ctx.project.path);
        const nextSteps = [
-               dirRelativePath
-                       ? ["Navigate to the new directory", `cd ${dirRelativePath}`]
-                       : [],
+               ...(dirRelativePath
+                       ? [["Navigate to the new directory", `cd ${dirRelativePath}`]]
+                       : []),
                [
                        "Run the development server",
                        quoteShellArgs([npm, "run", ctx.template.devScript ?? "start"]),
```

what do you think? 🙂

**Comment by DaniFoldi** [comment]
Hey @dario-piotrowicz,

Thanks, good idea! I've pushed a new commit changing it to your suggestion, I agree it's cleaner. Looked at the option of adding an explicit type annotation to prevent future issues, but ts couldn't validate all of them as `[string, string][]`, so it was going to require more type assertions than I'd be comfortable with for such a small change.

File: .changeset/quick-cheetahs-attack.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"create-cloudflare": patch
+---
+
+Skip empty next steps in the output summary. Previously, they would log undefined undefined.
```
Comment (dario-piotrowicz) [inline .changeset/quick-cheetahs-attack.md:5]: given the PR's change I would update the changeset 🙏 

something along these lines would be great 🙂
```suggestion
fix: make sure that using the current directory doesn't generate an `undefined undefined` log in the output summary.
```

the `fix:` prefix is as per our [changesets convention](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changeset-message-format)

**Review by dario-piotrowicz — APPROVED** [review]
Looks good to me, thanks a bunch @DaniFoldi ❤️ 🙂

**Comment by dario-piotrowicz** [comment]
> Thanks, good idea! I've pushed a new commit changing it to your suggestion, I agree it's cleaner. Looked at the option of adding an explicit type annotation to prevent future issues, but ts couldn't validate all of them as [string, string][], so it was going to require more type assertions than I'd be comfortable with for such a small change.

yeah I agree, if it's going to add unnecessary complexity it doesn't really feel worth it 🙂

**Review by jculvey — APPROVED** [review]

**Pull request merged**